### PR TITLE
Phase2-hgx293X Extend the possibility of new cells needed for rotated full wafers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Extend the possibility of new cells needed for rotated full wafers of HGCal

#### PR validation:

Tested using cfg's in Geometry/HGCalCommonData/test/python

#### if this PR is a backport please specify the original PR and why you need to report that PR:

Nothing special

### CODE REVIEW

File: Geometry/HGCalCommonData/plugins/DDHGCalCell.cc [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #include <string>
 #include <vector>
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (srimanob) [inline Geometry/HGCalCommonData/plugins/DDHGCalCell.cc:16]: Do you want to enable debug mode?

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/Utilities/interface/Exception.h"
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (srimanob) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc:7]: same here.

**Comment by srimanob** [comment]
Hi @bsunanda 

Could you please provide a bit more detail on the purpose of this PR?
- It seems you introduce `vtest`, this is for testing, right?
- Why touch also  `Geometry/HGCalCommonData/data/hgcalwafer/v15f/hgcal.xml`?

Thanks.

**Comment by bsunanda** [comment]
This is a step for future extension of rotated full wafers. The file Geometry/HGCalCommonData/data/hgcalwafer/v15f/hgcal.xml needs correction so that some of the test scripts could work. This file is needed only for tests and not for v15 definition

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/plugins/DDHGCalCell.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/DDHGCalCell.cc]: Done

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc]: Taken care of

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@AdrianoDee, @srimanob Please approve

**Comment by srimanob** [comment]
Hi @bsunanda 
Is this PR not related to testing only? Why do we see changes in all Phase-2 workflows?

**Comment by bsunanda** [comment]
No - it has some extension of cell structure. I tested with original as well as new ones. They give satisfactory results. Any small change in geometry makes perturbation in results

**Comment by bsunanda** [comment]
Please hold on - I am checking the code again

**Comment by bsunanda** [comment]
It was updated from a version CMSSW_12_1_X. So several new files get added. Close this in favor of #35997

**Pull request closed without merging**